Signal/src/ViewControllers/GifPicker/GifPickerViewController.swift

```diff
@@ -372,7 +372,7 @@ class GifPickerViewController: OWSViewController, UISearchBarDelegate, UICollect
                 owsFail("\(strongSelf.TAG) couldn't load asset.")
                 return
             }
-            let attachment = SignalAttachment(dataSource: dataSource, dataUTI: asset.rendition.utiType)
+            let attachment = SignalAttachment.imageAttachment(dataSource: dataSource, dataUTI: asset.rendition.utiType)
```

Comment:
The stricter rules around frameworks has exposed that we were using the wrong API here.

Reply:
Should we make that constructor private?

I agree this was wrong in principle, but it looks like this would have no negative side effects since the validation in `imageAttachment` should never apply to GIFs from gif picker, right?

Reply:
Which stricter rules, incidentally?  Around method visibility?

Reply:
> Should we make that constructor private?

done!

> Which stricter rules, incidentally? Around method visibility?

yup. It was "internal", but that breaks across modules.